extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/PeonLifecycleFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.k8s.overlord;
+
+import org.apache.druid.indexing.common.task.Task;
+
+public interface PeonLifecycleFactory
+{
+  KubernetesPeonLifecycle build(Task task);
```

Comment:
is there a reason we need this interface right now considering its in the k8s overlord extensions directory? feels pretty similar to the KubernetesPeonClient interface that was removed in this PR

Reply:
We are using it for testing, otherwise there is no way to mock/stub behavior for the PeonLifecycle in the unit tests.  Having to set up mocks for the classes/methods that the KubernetesPeonLifecycle object uses would tightly couple the KubernetesTaskRunner unit tests to the implementation of KubernetesPeonLifecycle.